docs/enhancements/install-sno-using-ibi.md

```diff
@@ -0,0 +1,133 @@
+---
+title: install-sno-using-ibi
+authors:
+  - "@adriengentil"
+creation-date: 2024-03-28
+last-updated: 2024-03-28
+---
+
+# Support Image-Based Installation in Assisted-Installer to install SNO
+
+## Summary
+
+Currently, installing an SNO cluster using the Assisted-Installer takes up to
+~40min using the current flow which installs OCP from the ground-up.
+
+We can reduce the installation time to ~10min by leveraging the Image-Based
+installation method, and as a way to increase the success rate of SNO
+installations as we restore previously working setup.
+
+This enhancement describes the changes to bring to the Assisted-Installer in
+order to perform SNO installation using the Image-Based installation method.
+
+## Motivation
+
+The Image-Based installation brings the following advantages:
+- Users will experience faster SNO installations
+- Increased success rate installations since we restore the state of a
+  previously working SNO setup
+
+### Goals
+
+- Provide an option in Assisted-Installer to allow a user to install a SNO with
+  the Image-Based Installation method
+
+### Non-Goals
+
+- Focus is on Assisted-Installer in SaaS environment, we don’t aim to support
+  other environments at this point
+
+## Proposal
+
+### As a user, I want to install a SNO using the IBI method
+
+#### REST API changes
+
+A new flag will be added to the cluster object [TBD], along with
+extra-parameters specific to IBI [TBD].
```

Comment:
I think it's worth coming up with at least a first pass on the API you'll add in the enhancement.

Reply:
I'll add it for sure once get some time, the rough idea I had was something like that in the cluster object:
```
{
  installation_method: "ibi",
  ibi_cluster_image: "quay.io/...." # allowed to be set only when installation_method is "ibi"
  openshift_version: nil, # because installation_method is set to "ibi"
  ocp_release_image: nil # because installation_method is set to "ibi"
}
```